### PR TITLE
fixed batch_size option processing only first batch

### LINKED ISSUE #30: batch_size acts like limit
When using ``batch_size``, not all records are being processed.  In fact only the number of records defined in ``batch_size`` is being processed.

I'm using a whitelisting strategy on an sqlite database.  The anonymization is defined like this:

```
require 'data-anonymization'
require 'sqlite3'

DataAnon::Utils::Logging.logger.level = Logger::INFO

database 'foobar' do

  strategy DataAnon::Strategy::Whitelist
  source_db :adapter => 'sqlite3', :database => 'foo.db'
  destination_db :adapter => 'sqlite3', :database => 'foo.anon.db'

  table "foo" do
    primary_key "id"
    batch_size 10

    whitelist "id"
    anonymize("title") { |field| field.value + "foo" }
  end

end
```

The table ``foo`` was created in source and destination database like this :
```
sqlite> CREATE TABLE foo(id INTEGER, title TEXT);
```

The table ``foo`` has 4999 records in the ``source_db``, and no records in the ``destination_db``.

When I run the anonymization script, only 10 records are created in the ``destination_db``.  I'm expecting that all 4999 records should appear in the destination db.

No errors are reported, the script output looks like:
```
[vagrant@localhost]$ ruby ruby_scripts/test.rb
I, [2016-01-19T15:11:19.867351 #8410]  INFO -- : Processing table foo records in batch size of 10
foo                  [     1/4999  ] ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉  0% 00:00:00
I, [2016-01-19T15:11:21.740611 #8410]  INFO -- : Fields missing the anonymization strategy
```

I tried with debug-level logging enabled, but no errors were shown then, either.

These are the gem versions installed:
```
*** LOCAL GEMS ***
activemodel (4.2.5)
activerecord (4.2.5)
activesupport (4.2.5)
arel (6.0.3)
bigdecimal (1.2.6)
bson (3.2.6, 1.12.5)
bson_ext (1.12.4)
builder (3.2.2)
composite_primary_keys (8.1.2)
data-anonymization (0.7.2)
hashie (3.4.3)
i18n (0.7.0)
io-console (0.4.3)
json (1.8.1)
minitest (5.4.3)
mongo (2.1.2)
parallel (1.6.1)
pg (0.18.4)
power_assert (0.2.2)
powerbar (1.0.16)
protected_attributes (1.1.3)
psych (2.0.8)
rake (10.4.2)
rdoc (4.2.0)
rgeo (0.5.2)
rgeo-geojson (0.4.2)
sqlite3 (1.3.11)
test-unit (3.0.8)
thor (0.19.1)
thread_safe (0.3.5)
tzinfo (1.2.2)
```

The version of ruby is 2.2.2p95.

### PR DESCRIPTION
**Pull request by janraasch**
fixes #30

Longwinded explanation:

There was some trouble with connecting to different databases through e.g.

```ruby
DataAnon::Utils::SourceDatabase.establish_connection @source_database
```

and

```ruby
DataAnon::Utils::DestinationDatabase.establish_connection @destination_database
```

In particular

the call to `DataAnon::Utils::DestinationDatabase.establish_connection` triggered by e.g.
`dest_record = dest_table.new dest_record_map, without_protection: true` in `#process_record` of the `Strategy` during processing the first batch caused the break-condition in `find_in_batches` of `ActiveRecord` to well *break* the loop and thus only the first batch would be processed.

To use different connection `ActiveRecord` likes to have *named* subclasses, which is why I created them in `BaseTable.create_table`.

Long story short: Fun debugging => it works :)

### CODE REVIEW

**Comment by janraasch** [comment]
Sorry for all the build and `--ammend` + `push --force` noise, but I do not quiet get the

```
./spec/utils/database_spec.rb:17
```

spec and how it works together with

```bash
rake empty_dest
```

**Comment by janraasch** [comment]
I am now down to the `ruby-head`-build failing bc. of an install error, see https://travis-ci.org/sunitparekh/data-anonymization/builds/106991655. Any idea what is going wrong here?

**Comment by sunitparekh** [comment]
First of all, thanks for helping in fixing the issue.
Give me couple of hours, I will look into it.
